### PR TITLE
#4385 Portfolio logic

### PR DESCRIPTION
**Pull request by desfero**
This pr connects business logic to the portfolio screen and portfolio section on the home screen.

**Tech notes**
The pr with wallet screen introduced some naming inconsistencies. Given we always call routing components screens therefore to keep consistency modules should be also called `<module>-screen`. 

**Testing notes**
Android: https://install.appcenter.ms/orgs/Neufund/apps/Neufund-Wallet-MVP/releases/16
IOS: https://install.appcenter.ms/orgs/Neufund/apps/Neufund-Wallet-MVP-iOS/releases/17

### CODE REVIEW

File: packages/shared-modules/src/modules/eto/utils.ts [hunk 1]
```diff
@@ -24,26 +21,26 @@ import {
...
+export const amendEtoToCompatibleFormat = <T extends TPartialEtoSpecData | TEtoDataWithCompany>(
+  eto: T,
+): T => ({
+  ...eto,
+  product: {
+    ...eto.product,
+    jurisdiction:
+      eto.product &&
+      eto.product.jurisdiction &&
+      (eto.product.jurisdiction.toUpperCase() as EJurisdiction),
```
Comment (Moejoe90) [inline packages/shared-modules/src/modules/eto/utils.ts:41]: I think we can already remove the upper case, But this is not related to this pr

File: packages/shared-modules/src/modules/investor-portfolio/sagas.ts [hunk 2]
```diff
@@ -135,28 +137,31 @@ export function* loadComputedContributionFromContract(
...
+export function* loadClaimables(): SagaGenerator<void> {
   const { logger, contractsService } = yield* neuGetBindings({
     logger: coreModuleApi.symbols.logger,
     contractsService: contractsModuleApi.symbols.contractsService,
   });
-  const user: IUser = yield select((state: TInvestorPortfolioModuleState) =>
-    authModuleAPI.selectors.selectUser(state),
-  );
+
+  const user = nonNullable(yield* select(authModuleAPI.selectors.selectUser));
```
Comment (Moejoe90) [inline packages/shared-modules/src/modules/investor-portfolio/sagas.ts:146]: :+1:

File: packages/wallet/app/modules/contracts/lib/utils.ts [hunk 3]
```diff
@@ -5,7 +5,9 @@ import { BigNumber } from "bignumber.js";
 import { utils } from "ethers";
 
 function bnToBne(number: BigNumber): utils.BigNumber {
-  return new utils.BigNumber(number.toString());
+  // note that bn.js do not support scientific notation
+  // therefore we need to force decimals in `toString`
+  return new utils.BigNumber(number.toString(10));
```
Comment (Moejoe90) [inline packages/wallet/app/modules/contracts/lib/utils.ts:10]: what does bne stand for exactly?

Comment (desfero) [reply packages/wallet/app/modules/contracts/lib/utils.ts:10]: It's shorthand of `bn ethers` as ethers is using `bn.js` where rest of our codebase `bignumber.js`

File: packages/wallet/app/modules/portfolio-screen/types.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+import { EIconType } from "components/shared/Icon";
+
+import { TToken } from "utils/types";
+
+import { setupPortfolioScreenModule } from "./module";
+
+export type TPortfolioScreenModuleState = TModuleState<typeof setupPortfolioScreenModule>;
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export type TAsset<T extends EquityToken = any, AT extends ECurrency = any> = {
```
[the remark this answers is not in the document]
Comment (desfero) [reply packages/wallet/app/modules/portfolio-screen/types.ts:13]: good point, we can just use `T extends EquityToken = EquityToken` and avoid any at all here. Thanks for pointing that
